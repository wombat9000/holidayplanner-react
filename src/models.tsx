export interface AppUser {
    displayName: string;
}