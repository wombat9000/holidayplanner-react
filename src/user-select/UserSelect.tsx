import * as React from 'react';
import {AppUser} from '../models';
import {Button} from '@material-ui/core';
import {UserContext} from '../UserContext';
import {Link} from 'react-router-dom';

const users: string[] = ['Wombat', 'Hedgehog', 'Moose'];

interface Props {
    onUserSelect: (user: AppUser) => void;
}

interface SelectorProps {
    displayName: string;
    handler: () => void;
}

export const UserSelector = (props: SelectorProps) => (
    <Button variant={'contained'} onClick={props.handler}>{props.displayName}</Button>
);

export const UserSelect = (props: Props) => {
    const selectUser = (name: string) => () => props.onUserSelect({displayName: name});
    const selectors = users.map((it, index: number) => <UserSelector key={index}
                                                                     displayName={it}
                                                                     handler={selectUser(it)}/>);

    return (
        <div>
            <UserContext.Consumer>
                {user => user ?
                    (<div>
                        <h1>Welcome {user.displayName}</h1>
                        <Link to={'/the-holiday'}>Go to Holiday</Link>
                    </div>) :
                    (<div>
                        <h1>UserSelect</h1>
                        {selectors}
                    </div>)
                }
            </UserContext.Consumer>
        </div>
    );
};
