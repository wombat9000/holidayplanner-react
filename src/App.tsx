import CssBaseline from '@material-ui/core/CssBaseline';
import * as React from 'react';
import './App.css';
import TopBar from './common/TopBar';
import {Route, Switch} from 'react-router';
import {UserSelect} from './user-select/UserSelect';
import {AppUser} from './models';
import {UserContext} from './UserContext';

class App extends React.Component<any, State> {
    constructor(props: any) {
        super(props);
        this.state = {
            title: 'Holiday',
        };

        this.setUser = this.setUser.bind(this);
    }

    private setUser(user: AppUser): void {
        this.setState({
            user: user
        });
    }

    public render() {
        return (
            <div className="App">
                <UserContext.Provider value={this.state.user}>
                    <CssBaseline/>
                    <TopBar title={this.state.title}/>
                    <Switch>
                        <Route render={() => (<UserSelect onUserSelect={this.setUser}/>)}/>
                    </Switch>
                </UserContext.Provider>
            </div>
        );
    }
}

interface State {
    title: string;
    user?: AppUser;
}

export default App;
