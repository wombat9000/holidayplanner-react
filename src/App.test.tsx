import * as React from 'react';
import App from './App';
import {mount, ReactWrapper, shallow, ShallowWrapper} from 'enzyme';
import {UserSelect} from './user-select/UserSelect';
import {AppUser} from './models';
import {MemoryRouter} from 'react-router';

describe('App', () => {
    let wrapper: ShallowWrapper;

    beforeEach(() => {
        wrapper = shallow(<App/>);
    });

    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    describe('default state', () => {
        it('should have Holiday as default title', () => {
            expect(wrapper.state('title')).toBe('Holiday');
        });

        it('should have no user by default', () => {
            expect(wrapper.state('user')).toBeUndefined();
        });
    });
});

xdescribe('Routing', () => {
    let wrapper: ReactWrapper;

    beforeEach(() => {
        wrapper = mount(<MemoryRouter><App/></MemoryRouter>);
    });

    describe('Default', () => {
        it('should update user state', () => {
            const userSelect = wrapper.find(UserSelect);
            const eventHandler: (user: AppUser) => void = userSelect.prop('onUserSelect');

            const someUser: AppUser = {
                displayName: 'The Wombat'
            };

            eventHandler(someUser);
            wrapper.update();
            expect(wrapper.find('h1.greeting').exists()).toBeTruthy();
        });
    });
});