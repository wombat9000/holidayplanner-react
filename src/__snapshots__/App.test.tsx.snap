// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <UNDEFINED
        value={undefined}
      >
        <WithStyles(CssBaseline) />
        <TopBar
          title="Holiday"
        />
        <Switch>
          <Route
            render={[Function]}
          />
        </Switch>
      </UNDEFINED>,
      "className": "App",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <WithStyles(CssBaseline) />,
          <TopBar
            title="Holiday"
          />,
          <Switch>
            <Route
              render={[Function]}
            />
          </Switch>,
        ],
        "value": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "title": "Holiday",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Route
              render={[Function]}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.provider),
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": [Circular],
          "Provider": [Circular],
          "_calculateChangedBits": null,
          "_changedBits": 0,
          "_changedBits2": 0,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": undefined,
          "_currentValue2": undefined,
          "_defaultValue": undefined,
        },
      },
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <UNDEFINED
          value={undefined}
        >
          <WithStyles(CssBaseline) />
          <TopBar
            title="Holiday"
          />
          <Switch>
            <Route
              render={[Function]}
            />
          </Switch>
        </UNDEFINED>,
        "className": "App",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <WithStyles(CssBaseline) />,
            <TopBar
              title="Holiday"
            />,
            <Switch>
              <Route
                render={[Function]}
              />
            </Switch>,
          ],
          "value": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "title": "Holiday",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Route
                render={[Function]}
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.provider),
          "_context": Object {
            "$$typeof": Symbol(react.context),
            "Consumer": [Circular],
            "Provider": [Circular],
            "_calculateChangedBits": null,
            "_changedBits": 0,
            "_changedBits2": 0,
            "_currentRenderer": null,
            "_currentRenderer2": null,
            "_currentValue": undefined,
            "_currentValue2": undefined,
            "_defaultValue": undefined,
          },
        },
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
